### PR TITLE
Update the internships page to reflect Outreachy end and GSoD start

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description
Since Outreachy (summer) and GSoC are ending and GSoD has started, this PR updates the internship pages, both current and history, to reflect these changes.
- Mentions GSoD result announcement
- Retires Outreachy and GSoC to history page
- Adds GSoD to the history page

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by Dhruvi16 — APPROVED** [review]

File: content/internships/contents.lr [hunk 1]
```diff
@@ -12,24 +12,10 @@ Creative Commons is currently participating in the current rounds of the followi
 
 ### Google Season of Docs
 
-- [Program website](https://developers.google.com/season-of-docs/)
-- CC's [Project Ideas](/internships/project-ideas/)
-
-If you're interested in this program, please review the resources below and come talk to us on Slack in `#cc-gsoc-outreachy`.
-
-### Google Summer of Code
```
Comment (kgodey) [inline content/internships/contents.lr:20]: GSoC doesn't end until the end of the month, only Outreachy has ended.

File: content/internships/history/contents.lr [hunk 2]
```diff
@@ -6,10 +6,16 @@ title: Open Source Internships: History
...
+## Google Season of Docs
+
+CC has participated in Google Season of Docs for the first time in 2020.
+
 ## Google Summer of Code
 
 CC has participated in Google Summer of Code eight times before 2020. See below for more information about our projects and participation for each year.
 
+* [Google Summer of Code 2020](https://summerofcode.withgoogle.com/organizations/5742311717208064/)
+    * [Open Source Blog posts](/blog/categories/gsoc-2020/)
```
Comment (kgodey) [inline content/internships/history/contents.lr:18]: please also link to https://creativecommons.org/2020/05/11/welcome-interns-google-summer-of-code-and-outreachy/

File: content/internships/history/contents.lr [hunk 3]
```diff
@@ -29,6 +35,6 @@ CC has participated in Google Summer of Code eight times before 2020. See below
 
 ## Outreachy
 
-CC participated in the Dec 2019 to Mar 2020 round of Outreachy.  
+CC participated in the May 2020 to December 2020 round of Outreachy.
```
Comment (kgodey) [inline content/internships/history/contents.lr:38]: We participated in both rounds, this is the history page so it should include both.

File: content/internships/contents.lr (hunk 1 above)
Comment (dhruvkb) [reply content/internships/contents.lr:20]: Noted, my mistake. I falsely assumed that they had identical timelines.

Comment (dhruvkb) [reply content/internships/contents.lr:20]: Resolved in 99fe0d3.

File: content/internships/history/contents.lr (hunk 3 above)
Comment (dhruvkb) [reply content/internships/history/contents.lr]: Resolved in f00524f.

File: content/internships/history/contents.lr (hunk 2 above)
Comment (dhruvkb) [reply content/internships/history/contents.lr:26]: Resolved in f00524f.

**Pull request merged**